### PR TITLE
build: bump `fuel-vm` to `0.4` and `fuel-tx` to `0.5`

### LINKED ISSUE #84: Update `fuel-tx` to v0.5.0 and `fuel-vm` to v0.4.0
This may or may not break some things inside `fuels-rs`.

### PR DESCRIPTION
**Pull request by vnepveu**
Use the newer versions of `fuel-tx` and `fuel-vm` in dependencies, and adapt the code accordingly
This PR closes #84.

### CODE REVIEW

**Comment by vnepveu** [comment]
Currently struggling with using `fuel-tx::{Transaction,Receipts}` properly,I keep getting this error:
```bash
```

**Comment by vnepveu** [comment]
The version bumps now makes 5 tests fail (others run well):
```
test call_with_empty_return ... FAILED
test abigen_different_structs_same_arg_name ... FAILED
test example_workflow ... FAILED
test call_with_structs ... FAILED
test type_safe_output_values ... FAILED
```
They fail with `Custom { kind: Other, error: "Response errors; The provided identifier is invalid!" }`, which occurs on contract calls (in `script.rs`).

File: fuels-abigen-macro/Cargo.toml
Comment (adlerjohn) [inline fuels-abigen-macro/Cargo.toml:24]: You probably want to bump `fuel-core` in line with `fuel-gql-client`, since they're versioned in lockstep.

File: fuels-contract/Cargo.toml
Comment (adlerjohn) [inline fuels-contract/Cargo.toml:16]: Ditto.

**Review by adlerjohn — COMMENTED** [review]
While you're here, can you also sort the dependencies alphabetically?

File: fuels-abigen-macro/Cargo.toml
Comment (vnepveu) [reply fuels-abigen-macro/Cargo.toml:24]: :facepalm: it's obvious, `gql-client` is litteraly part of `fuel-core`. And that fixes the tests, thanks a lot!

**Review by adlerjohn — COMMENTED** [review]
Generally lgtm, but will defer to @digorithm. @vnepveu can you file an issue to track adding initial storage slots in the SDK 1) manually, and 2) automatically based on some output by the compiler (latter being blocked by it being implemented in the compiler obviously)?

**Review by digorithm — APPROVED** [review]
LGTM!

**Pull request merged**
